micro_manager/interpolation.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+from sklearn.neighbors import NearestNeighbors
+
+
+class Interpolation:
+    def __init__(self, logger):
+
+        self._logger = logger
+
+    def get_nearest_neighbor_indices_local(
+        self,
+        all_local_coords,
+        inter_point,
+        k: int,
+        inter_point_is_neighbor: bool = False,
+    ) -> np.ndarray:
+        """
+        Get the indices of the k nearest neighbors of a point in a list of coordinates.
+        Note: It can be chosen whether the point itself is considered as a neighbor or not.
+        Args:
+            all_local_coords: list
+                List of coordinates of all points.
+            inter_point:
+                Coordinates of the point for which the neighbors are to be found.
+            k: int
+            inter_point_is_neighbor: bool, optional
+                Decide whether the interpolation point is considered as its own neighbor.
+                Defaults to False.
```

Comment:
What is the reasoning behind providing this option? Does it change the accuracy of the interpolation?

Reply:
If the distance is zero, inverse distance weighting will have a zero division but there is also no reason to interpolate as information at the interpolation point exists. In the case of crash interpolation, this case should never occur as there is no information and that's why interpolation is performed. The option can be removed.